src/rules/no-unnormalized-keys.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview Rule to detect unnormalized keys in JSON.
+ * @author Bradley Meck Farias
+ */
+
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			description: "Disallow JSON keys that are not normalized",
+		},
+
+		messages: {
+			unnormalizedKey: "Unnormalized key found.",
+		},
+
+		schema: {
+			type: "array",
+			minItems: 0,
+			maxItems: 1,
+			items: {
+				enum: ["NFC", "NFD", "NFKC", "NFKD"],
+			},
+		},
```

Comment:
```suggestion
		schema: [
			{
				enum: ["NFC", "NFD", "NFKC", "NFKD"],
			},
		],
```

This can be simplified by using the shorthand form (https://eslint.org/docs/latest/extend/custom-rules#options-schemas, the first format).